src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -240,18 +239,27 @@ abstract class ErgoNodeViewHolder[State <: ErgoState[State]](settings: ErgoSetti
         log.error("Reporting modifier failed", ex)
         f
       case (success@Success(updateInfo), modToApply) =>
+        def reportInvalidModifier(ex: Throwable): Try[UpdateInformation] =
+          history.reportModifierIsInvalid(modToApply, progressInfo).map { case (newHis, newProgressInfo) =>
+            context.system.eventStream.publish(SemanticallyFailedModification(modToApply, ex))
+            UpdateInformation(newHis, updateInfo.state, Some(modToApply), Some(newProgressInfo), updateInfo.suffix)
+          }
         if (updateInfo.failedMod.isEmpty) {
           updateInfo.state.applyModifier(modToApply, estimatedTip())(lm => pmodModify(lm.pmod, local = true)) match {
             case Success(stateAfterApply) =>
               history.reportModifierIsValid(modToApply).map { newHis =>
                 context.system.eventStream.publish(SemanticallySuccessfulModifier(modToApply))
                 UpdateInformation(newHis, stateAfterApply, None, None, updateInfo.suffix :+ modToApply)
               }
-            case Failure(e) =>
-              history.reportModifierIsInvalid(modToApply, progressInfo).map { case (newHis, newProgressInfo) =>
-                context.system.eventStream.publish(SemanticallyFailedModification(modToApply, e))
-                UpdateInformation(newHis, updateInfo.state, Some(modToApply), Some(newProgressInfo), updateInfo.suffix)
+            case Failure(ex: MalformedModifierError) if ex.modifierTypeId == Transaction.ModifierTypeId =>
```

Comment:
It works like this : 
 - in either way : `modToApply` is reported as invalid
 - if `MalformedModifierError` with `TxType` is catched, we managed to find the invalid TX here https://github.com/ergoplatform/ergo/pull/1803/files#diff-85a518f9e45c7c4c4119bd8b54f22b95446a2c2c24b5b8328e1c6c4b7bbca4bdR93-R101
 - otherwise we just report invalid `modToApply`

Reply:
Not so simple, as  `history.reportModifierIsInvalid` is expecting block ,not transaction

Reply:
We call `reportModifierIsInvalid(modToApply)` in both cases, just in case of `MalformedModifierError` with `TxType` modifier we `EliminateTransactions` so it gets removed from MemPool